src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryMapping.cs

```diff
@@ -208,8 +210,20 @@ void Update<T>(Dictionary<string, DynamicQueryMappingItem> fields, Dictionary<st
                 }
             }
         }
+        
+        public static DynamicQueryMapping Create(IndexQueryServerSide query, SearchEngineType searchEngineType)
+        {
+            var result = CreateInternal(query);
+            
+            result.SearchEngineType = searchEngineType;
+
+            if (query.Metadata.HasVectorSearch)
+                result.SearchEngineType = SearchEngineType.Corax;
```

Comment:
Shouldn't we make this check outside? It's a bit tricky that `Create()` method accepting search engine type as parameter will return a result with different search engine set.

Reply:
rename the parameter to `defaultSearchEngineType`

Reply:
Changed